test/e2e/storage/testsuites/multireader_stress.go

```diff
@@ -0,0 +1,256 @@
+	stopChs []chan struct{}
+}
+
+var _ TestSuite = &multiReaderStressTestSuite{}
+
+// InitStressTestSuite returns multiReaderStressTestSuite that implements TestSuite interface
+func InitStressTestSuite() TestSuite {
+	return &multiReaderStressTestSuite{
+		tsInfo: TestSuiteInfo{
+			Name:       "multireader-stress",
+			FeatureTag: "[Feature: VolumeStress]",
+			TestPatterns: []testpatterns.TestPattern{
+				testpatterns.DefaultFsDynamicPV,
+			},
+		},
+	}
+}
+
+func (t *multiReaderStressTestSuite) GetTestSuiteInfo() TestSuiteInfo {
+	return t.tsInfo
+}
+
+func (t *multiReaderStressTestSuite) SkipRedundantSuite(driver TestDriver, pattern testpatterns.TestPattern) {}
+
+func (t *multiReaderStressTestSuite) DefineTests(driver TestDriver, pattern testpatterns.TestPattern) {
+	var (
+		dInfo = driver.GetDriverInfo()
+		cs    clientset.Interface
+	)
+
+	ginkgo.BeforeEach(func() {
+		// Check preconditions.
+		_, ok := driver.(DynamicPVTestDriver)
+		if !ok {
+			e2eskipper.Skipf("Driver %s doesn't support dynamic PV mode -- skipping", dInfo.Name)
+		}
+
+		_, ok = driver.(ReadOnlyPVTestDriver)
+		if !ok {
+			e2eskipper.Skipf("Driver %s doesn't support readonly PV mode -- skipping", dInfo.Name)
+		}
+	})
+
+	// This intentionally comes after checking the preconditions because it
+	// registers its own BeforeEach which creates the namespace. Beware that it
+	// also registers an AfterEach which renders f unusable. Any code using
+	// f must run inside an It or Context callback.
+	f := framework.NewDefaultFramework("multireader-stress")
+
+	init := func() *multiReaderStressTest {
+		cs = f.ClientSet
+		l := &multiReaderStressTest{}
+
+		// Now do the more expensive test initialization.
+		l.config, l.driverCleanup = driver.PrepareTest(f)
+		l.intreeOps, l.migratedOps = getMigrationVolumeOpCounts(f.ClientSet, dInfo.InTreePluginName)
+		l.stopChs = []chan struct{}{}
+
+		return l
+	}
+
+	cleanup := func(l *multiReaderStressTest) {
+		var errs []error
+
+		framework.Logf("Stopping and waiting for all test routines to finish")
+		for _, stopCh := range l.stopChs {
+			close(stopCh)
+		}
+		l.wg.Wait()
+
+		framework.Logf("Deleting reader PV and PVC")
+		readerVolumeCleanupErrs := e2epv.PVPVCCleanup(cs, f.Namespace.Name, l.readerPV, l.readerPVC)
+		errs = append(errs, readerVolumeCleanupErrs...)
+
+		for _, pod := range l.readerPods {
+			framework.Logf("Deleting pod %v", pod.Name)
+			err := e2epod.DeletePodWithWait(cs, pod)
+			errs = append(errs, err)
+		}
+
+		if l.writerResource != nil {
+			framework.Logf("Deleting writer PV and PVC")
+			errs = append(errs, l.writerResource.CleanupResource())
+		}
+
+		errs = append(errs, tryFunc(l.driverCleanup))
+		framework.ExpectNoError(errors.NewAggregate(errs), "while cleaning up resource")
+		validateMigrationVolumeOpCounts(f.ClientSet, dInfo.InTreePluginName, l.intreeOps, l.migratedOps)
+	}
+
+	ginkgo.It("multiple pods should access a single read-only volume repeatedly [Slow] [Serial]", func() {
```

Comment:
This test should be skipped for plugins that don't support ROX. Maybe a new capability is required.